src/schema/entities.yaml

```
@@ -0,0 +1,127 @@
+                the day after) those can be grouped in one session.
+                Defining multiple sessions is appropriate when several identical
+                or similar data acquisitions are planned and performed on all
+                -or most- subjects, often in the case of some intervention between
+                sessions (e.g., training)."
+  format: label
+task:
+  name: Task
+  format: label
+  description: "Each task has a unique label that MUST only consist of letters
+                and/or numbers (other characters, including spaces and
+                underscores, are not allowed).
+                Those labels MUST be consistent across subjects and sessions."
+acq:
+  name: Acquisition
+  description: "The OPTIONAL acq-<label> key/value pair corresponds to a custom
+                label the user MAY use to distinguish a different set of
+                parameters used for acquiring the same modality.
+                For example this should be used when a study includes two T1w
+                images - one full brain low resolution and and one restricted
+                field of view but high resolution. In such case two files could
+                have the following names: sub-01_acq-highres_T1w.nii.gz and
+                sub-01_acq-lowres_T1w.nii.gz, however the user is free to choose
+                any other label than highres and lowres as long as they are
+                consistent across subjects and sessions. In case different
+                sequences are used to record the same modality (e.g. RARE and
+                FLASH for T1w) this field can also be used to make that distinction.
+                At what level of detail to make the distinction (e.g. just between
+                RARE and FLASH, or between RARE, FLASH, and FLASHsubsampled)
+                remains at the discretion of the researcher."
+  format: label
+ce:
+  name: Contrast Enhancing Agent
+  description: "Similarly the OPTIONAL ce-<label> key/value can be used to
+                distinguish sequences using different contrast enhanced images.
+                The label is the name of the contrast agent.
+                The key ContrastBolusIngredient MAY be also be added in the JSON
+                file, with the same label."
+  format: label
+rec:
+  name: Reconstruction
+  description: "Similarly the OPTIONAL rec-<label> key/value can be used to
+                distinguish different reconstruction algorithms (for example
+                ones using motion correction)."
+  format: label
+dir:
+  name: Phase-Encoding Direction
+  description: "Similarly the OPTIONAL dir-<label> key/value can be used to
+                distinguish different phase-encoding directions."
+  format: label
+run:
+  name: Run
+  description: "If several scans of the same modality are acquired they MUST be
+                indexed with a key-value pair: `_run-1`, `_run-2`, `_run-3` etc.
+                (only integers are allowed as run labels).
```

Do we require that run indices start at 1?  Should we say that explicitly?

```suggestion
                (only natural numbers are allowed as run labels).
```

---

Since this is a change to the specification, should it be a separate issue? On the other hand, it's such a small change I don't know if it needs to be separated out.

---

Fair point. I'll lodge a comment elsewhere.

---

Unfortunately, natural numbers doesn't have a universally agreed meaning (some include 0, some don't) so this would require us to define it.

It might be better to say "only numeric values", as this is what is actually validated. It's equivalent to non-negative integers with optional zero-padding, but that seems like a cumbersome phrasing.

---

Good point.  Please see #535.